src/app/InteractionModelEngine.cpp

```diff
@@ -324,6 +325,17 @@ void InteractionModelEngine::OnDone(ReadHandler & apReadObj)
     mReportingEngine.ResetReadHandlerTracker(&apReadObj);
 
     mReadHandlers.ReleaseObject(&apReadObj);
+
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS && CHIP_CONFIG_SUBSCRIPTION_TIMEOUT_RESUMPTION
+    if (!mSubscriptionResumptionScheduled && HasSubscriptionsToResume())
+    {
+        mSubscriptionResumptionScheduled  = true;
+        auto timeTillNextResubscriptionMs = ComputeTimeTillNextSubscriptionResumption();
+        mpExchangeMgr->GetSessionManager()->SystemLayer()->StartTimer(System::Clock::Seconds32(timeTillNextResubscriptionMs),
```

Comment:
You are initiating a Seconds32 with a variable named `Ms`.

Add a unit of measure extension to ComputeTimeTillNextSubscriptionResumption (is it seconds or MS? should it maybe return a Seconds32 or similar instead of a raw value?

Reply:
This follows the same convention as `ReadClient::ComputeTimeTillNextSubscription` but I've taken your advice and renamed the function to be clearer, and corrected the variable name.